pkg/controller/iotconfig/iotconfig_controller.go

```diff
@@ -231,6 +235,11 @@ func (r *ReconcileIoTConfig) updateStatus(ctx context.Context, original *iotv1al
 
 	syncConfigCondition(&config.Status)
 
+	if config.Status.Phase != original.Status.Phase {
+		// record event that the
+		r.recorder.Eventf(config, corev1.EventTypeNormal, "Activated", "IoT infrastructure successfully activated")
```

Comment:
Code looks surprising.  config.Status.Phase != original.Status.Phase does not imply to me "Activated".   Either comment to explain why this so, or perhaps the event should capture the state change?

Reply:
You are right, this was not correct. Fixed. Thanks!